elephant/test/test_spectral.py

```diff
@@ -388,6 +390,537 @@ def test_multitaper_psd_input_types(self):
+                                   np.diagonal(cross_spec).T.real.magnitude,
+                                   rtol=0.01,
+                                   atol=0.01)
+
+    def test_segmented_multitaper_cross_spectrum_input_types(self):
+        # generate a test data
+        data_length = 5000
+        sampling_period = 0.001
+        signal_freq = 100.0
+        noise = np.random.normal(size=(2, data_length))
+        time_points = np.arange(0, data_length * sampling_period,
+                                sampling_period)
+        signal_x = np.sin(2 * np.pi * signal_freq * time_points) + noise[0]
+        signal_y = np.cos(2 * np.pi * signal_freq * time_points) + noise[1]
+        data = n.AnalogSignal(np.vstack([signal_x, signal_y]).T,
+                              sampling_period=sampling_period * pq.s,
+                              units='mV')
+
+        # frequency resolution as an integer
+        freq_res_int = 1
+        freq_res_hz = 1 * pq.Hz
+
+        freqs_int, cross_spec_int = \
+            elephant.spectral.segmented_multitaper_cross_spectrum(
+                data, frequency_resolution=freq_res_int)
+
+        freqs_hz, cross_spec_hz = \
+            elephant.spectral.segmented_multitaper_cross_spectrum(
+                data, frequency_resolution=freq_res_hz)
+
+        np.testing.assert_array_equal(freqs_int, freqs_hz)
+        np.testing.assert_array_equal(cross_spec_int, cross_spec_hz)
+
+
+class MultitaperCoherenceTestCase(unittest.TestCase):
+    def test_multitaper_coherence_input_types(self):
+        # Generate dummy data
+        data_length = 10000
+        sampling_period = 0.001
+        signal_freq = 100.0
+        np.random.seed(123)
+        noise = np.random.normal(size=(2, data_length))
+        time_points = np.arange(0, data_length * sampling_period,
+                                sampling_period)
+        # Signals are designed to have coherence peak at `signal_freq`
+        arr_signal_i = np.sin(2 * np.pi * signal_freq * time_points) + noise[0]
+        arr_signal_j = np.cos(2 * np.pi * signal_freq * time_points) + noise[1]
+
+        fs = 1000 * pq.Hz
+        anasig_signal_i = neo.core.AnalogSignal(arr_signal_i,
+                                                sampling_rate=fs,
+                                                units=pq.mV)
+        anasig_signal_j = neo.core.AnalogSignal(arr_signal_j,
+                                                sampling_rate=fs,
+                                                units=pq.mV)
+
+        arr_f, arr_coh, arr_phi = elephant.spectral.multitaper_coherence(
+            arr_signal_i, arr_signal_j, fs=fs)
+        anasig_f, anasig_coh, anasig_phi = \
+            elephant.spectral.multitaper_coherence(anasig_signal_i,
+                                                   anasig_signal_j)
+
+        np.testing.assert_array_equal(arr_f, anasig_f)
+        np.testing.assert_allclose(arr_coh, anasig_coh, atol=1e-8)
```

Comment:
This unit tests fails: 

```
=================================== FAILURES ===================================
______ MultitaperCoherenceTestCase.test_multitaper_coherence_input_types _______

self = <elephant.test.test_spectral.MultitaperCoherenceTestCase testMethod=test_multitaper_coherence_input_types>

    def test_multitaper_coherence_input_types(self):
        # Generate dummy data
        data_length = 10000
        sampling_period = 0.001
        signal_freq = 100.0
        np.random.seed(123)
        noise = np.random.normal(size=(2, data_length))
        time_points = np.arange(0, data_length * sampling_period,
                                sampling_period)
        # Signals are designed to have coherence peak at `signal_freq`
        arr_signal_i = np.sin(2 * np.pi * signal_freq * time_points) + noise[0]
        arr_signal_j = np.cos(2 * np.pi * signal_freq * time_points) + noise[1]
    
        fs = 1000 * pq.Hz
        anasig_signal_i = neo.core.AnalogSignal(arr_signal_i,
                                                sampling_rate=fs,
                                                units=pq.mV)
        anasig_signal_j = neo.core.AnalogSignal(arr_signal_j,
                                                sampling_rate=fs,
                                                units=pq.mV)
    
        arr_f, arr_coh, arr_phi = elephant.spectral.multitaper_coherence(
            arr_signal_i, arr_signal_j, fs=fs)
        anasig_f, anasig_coh, anasig_phi = \
            elephant.spectral.multitaper_coherence(anasig_signal_i,
                                                   anasig_signal_j)
    
        np.testing.assert_array_equal(arr_f, anasig_f)
>       np.testing.assert_allclose(arr_coh, anasig_coh, atol=1e-8)

elephant/test/test_spectral.py:819: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

args = (<function assert_allclose.<locals>.compare at 0x7faa8a8edf70>, array([0.00078479, 0.12790763, 0.12810889, ..., 0.1939...y([0.00078479, 0.12790763, 0.12810889, ..., 0.19397819, 0.189415  ,
       0.27107906], dtype=float32) * dimensionless)
kwds = {'equal_nan': True, 'err_msg': '', 'header': 'Not equal to tolerance rtol=1e-07, atol=1e-08', 'verbose': True}

    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError: 
E           Not equal to tolerance rtol=1e-07, atol=1e-08
E           
E           Mismatched elements: 35 dimensionless / 5001 (0.7 dimensionless%)
E           Max absolute difference: 1.1920929e-07
E           Max relative difference: 2.1323707e-07
E            x: array([0.000785, 0.127908, 0.128109, ..., 0.193978, 0.189415, 0.271079],
E                 dtype=float32)
E            y: Quantity([0.000785, 0.127908, 0.128109, ..., 0.193978, 0.189415, 0.271079],
E                    dtype=float32)

/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/contextlib.py:79: AssertionError
```

env:
```
Package            Version   Editable project location
------------------ --------- -----------------------------------
attrs              22.2.0
certifi            2022.12.7
charset-normalizer 3.0.1
coverage           6.5.0
coveralls          3.3.1
docopt             0.6.2
elephant           0.12.0b1  /home/runner/work/elephant/elephant
exceptiongroup     1.1.0
h5py               3.8.0
idna               3.4
iniconfig          2.0.0
Jinja2             3.1.2
joblib             1.2.0
MarkupSafe         2.1.2
mpi4py             3.1.4
neo                0.12.0
nixio              1.5.3
numpy              1.24.2
packaging          23.0
pandas             1.5.3
patsy              0.5.3
pip                23.0
pluggy             1.0.0
pytest             7.2.1
pytest-cov         4.0.0
python-dateutil    2.8.2
pytz               2022.7.1
quantities         0.14.1
requests           2.28.2
scikit-learn       1.2.1
scipy              1.10.0
setuptools         58.1.0
six                1.16.0
statsmodels        0.13.5
threadpoolctl      3.1.0
tomli              2.0.1
tqdm               4.64.1
urllib3            1.26.14
Python 3.9.16
```

Reply:
We ran the unit test locally and with even higher precision and it still passed. We can circumvent the problem by reducing the precision to, e.g. 1e-6. Is this an acceptable solution?

Reply:
Hey, so far I also failed at reproducing this behavior locally. 
It seems this is not a major problem with your implementation.

At the moment i see no other option than to reduce the precision. (also 1e-6 seems acceptable to me)